AsyncDisplayKit/ASCollectionNode.h

```diff
@@ -131,20 +241,346 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)reloadDataWithCompletion:(nullable void (^)())completion;
 
+#pragma mark - Querying Data
+
+/**
+ * Retrieves the number of items in the given section.
+ *
+ * @param section The section.
+ *
+ * @return The number of items.
+ */
+- (NSInteger)numberOfItemsInSection:(NSInteger)section AS_WARN_UNUSED_RESULT;
+
+/**
+ * The number of sections.
+ */
+@property (nonatomic, readonly) NSInteger numberOfSections;
+
+/**
+ * Similar to -visibleCells.
+ *
+ * @return an array containing the nodes being displayed on screen.
+ */
+- (NSArray<__kindof ASCellNode *> *)visibleNodes AS_WARN_UNUSED_RESULT;
+
+/**
+ * Retrieves the node for the item at the given index path.
+ *
+ * @param indexPath The index path of the requested item.
+ *
+ * @return The node for the given item, or @c nil if no item exists at the specified path.
+ */
+- (nullable ASCellNode *)nodeForItemAtIndexPath:(NSIndexPath *)indexPath AS_WARN_UNUSED_RESULT;
+
+/**
+ * Retrieve the index path for the item with the given node.
+ *
+ * @param cellNode A node for an item in the collection node.
+ *
+ * @return The indexPath for this item.
+ */
+- (nullable NSIndexPath *)indexPathForNode:(ASCellNode *)cellNode AS_WARN_UNUSED_RESULT;
+
+
+/**
+ * Retrieves the context object for the given section, as provided by the data source in
+ * the @c collectionNode:contextForSection: method.
+ *
+ * @param section The section to get the context for.
+ *
+ * @return The context object, or @c nil if no context was provided.
+ *
+ * TODO: This method currently accepts @c section in the _view_ index space, but it should
+ *   be in the node index space. To get the context in the view index space (e.g. for subclasses
+ *   of @c UICollectionViewLayout, the user will call the same method on @c ASCollectionView.
+ */
+- (nullable id<ASSectionContext>)contextForSection:(NSInteger)section AS_WARN_UNUSED_RESULT;
+
+@end
+
+@interface ASCollectionNode (Deprecated)
+
 /**
  * Reload everything from scratch, destroying the working range and all cached nodes.
  *
  * @warning This method is substantially more expensive than UICollectionView's version.
+ *
+ * @deprecated This method is deprecated in 2.0. Use @c reloadDataWithCompletion: instead.
+ */
+- (void)reloadData ASDISPLAYNODE_DEPRECATED;
+
+/**
+ * Reload everything from scratch, destroying the working range and all cached nodes.
+ *
+ * @warning This method is substantially more expensive than UICollectionView's version.
+ *
+ * @deprecated This method is deprecated in 2.0. Use @c reloadDataWithCompletion: and
+ *   then @c waitUntilAllUpdatesAreCommitted instead.
+ */
+- (void)reloadDataImmediately ASDISPLAYNODE_DEPRECATED;
+
+@end
+
+/**
+ * This is a node-based UICollectionViewDataSource.
+ */
+@protocol ASCollectionDataSource <ASCommonCollectionDataSource>
```

Comment:
This (and `ASCollectionDelegate` `ASTableDataSource` and `ASTableDelegate`) was kept as-is, but moved from the `View.h` file to the `Node.h` file.